src/main/kotlin/com/exactpro/th2/kafka/client/Main.kt

```diff
@@ -63,53 +66,98 @@ fun main(args: Array<String>) {
-
-                val bookName = message.bookName
-                if (bookName.isNotEmpty() && bookName != factory.boxConfiguration.bookName) {
-                    val errorText = "Expected bookName: '${factory.boxConfiguration.bookName}', actual '$bookName' in message ${message.id.logId}"
-                    LOGGER.error { errorText }
-                    eventSender.onEvent(errorText, "Error", status = Event.Status.FAILED)
-                    continue
+        val subscribe = if (config.useTransport) {
+            val transportRouter = factory.transportGroupBatchRouter
+            val messageProcessor = TransportRawMessageProcessor(config.batchSize, config.timeSpan, config.timeSpanUnit, factory.boxConfiguration.bookName, config.aliasToSessionGroup) {
+                LOGGER.trace { "Sending batch with ${it.groups.size} groups to MQ." }
+                it.runCatching(transportRouter::send).onFailure { e -> LOGGER.error(e) { "Could not send message batch to MQ: $it" } }
+            }.apply { resources += "message processor" to ::close }
+
+            val connection = TransportKafkaConnection(config, factory, messageProcessor, eventSender, KafkaClientsFactory(config))
+                .apply { resources += "kafka connection" to ::close }
+
+            val transportListener: (DeliveryMetadata, GroupBatch) -> Unit = { metadata, batch ->
+                LOGGER.trace { "Transport batch with ${batch.groups.size} groups received from MQ"}
+                for (group in batch.groups) {
+                    if (group.messages.size != 1) {
+                        val errorText = "Transport message group must contain only one message. Consumer tag ${metadata.consumerTag}"
+                        LOGGER.error { errorText }
+                        eventSender.onEvent(errorText, "SendError")
+                        break
+                    }
+
+                    val message = group.messages[0]
+                    LOGGER.trace { "Message ${message.id.logId} extracted from batch." }
+
+                    if (message !is RawMessage) {
+                        val errorText = "Transport message ${message.id.logId} is not a raw message. Consumer tag ${metadata.consumerTag}"
+                        LOGGER.error { errorText }
+                        eventSender.onEvent(errorText, "SendError")
+                        break
+                    }
+
+                    runCatching {
+                        connection.publish(message)
+                    }.onFailure {
+                        val errorText = "Could not publish message ${message.id.logId}. Consumer tag ${metadata.consumerTag}"
+                        LOGGER.error(it) { errorText }
+                        eventSender.onEvent(errorText, "SendError", message.id.toProto(batch), it)
+                    }
+                }
+            }
+            { transportRouter.subscribeAll(transportListener, INPUT_QUEUE_ATTRIBUTE) }
```

Comment:
Use different set of attributes for transport and protobuf

Reply:
Attributes are different: transport router adds "transport-group" during subscribe and raw batch router adds "raw" attribute.

Reply:
O, we use deprecated messageRouterRawBatch which add raw attribute. OK